lib/cog/command/option_interpreter.ex

```diff
@@ -1,20 +1,20 @@
 defmodule Cog.Command.OptionInterpreter do
   require Logger
-  alias Cog.Command.CommandCache
+  alias Cog.Models.CommandOption
   alias Piper.Command.Ast
 
-  @truthy_values ["1", "t", "true", "y", "yes", "on"]
+  require Logger
+
+  @truthy_values ["true", "t", "y", "yes", "on"]
 
-  def initialize(%Ast.Invocation{}=command, raw) do
-    case CommandCache.fetch_options(command, :prepared) do
-      :not_found ->
-        :not_found
-      {:ok, defs} ->
-        case interpret(defs, raw, [], %{}) do
+  def initialize(%Ast.Invocation{args: args, meta: command_model}=invocation) do
+    case Enum.reduce(command_model.options, %{}, &(prepare_option(&1, &2))) do
```

Comment:
Might be a good place to use `with`

```
defs = Enum.reduce(command_model.options, %{}, &(prepare_option(&1, &2)))

with {:ok, options, args} <- interpret(defs, args, [], %{}),
     :ok <- check_required_options(defs, options) do
  options = set_defaults(invocation, options)
  {:ok, options, args}
end
```

Also not sure if `defs` can ever not be a map so that first case might be unnecessary.